tests/tests_pytorch/utilities/test_imports.py

```diff
@@ -67,3 +75,54 @@ def test_imports():
         assert not _POPTORCH_AVAILABLE
     else:
         assert _POPTORCH_AVAILABLE
+
+
+def _shortcut_patch(orig_fn, shortcut_case, attr_names=None):
+    def new_fn(*args, **kwargs):
+        if attr_names is not None:
+            self = args[0]
+            values = tuple(getattr(self, attr_name) for attr_name in attr_names)
+        else:
+            values = args
+        match = True
+        for value, case in zip(values, shortcut_case):
+            if value != case:
+                match = False
+                break
+        if match:
+            return False
+        return orig_fn(*args, **kwargs)
+
+    return new_fn
+
+
+@pytest.mark.parametrize(
+    ["patch_name", "new_fn", "to_import"],
+    [
+        ("torch.distributed.is_available", _shortcut_patch(is_available, ()), "pytorch_lightning"),
+        (
+            "lightning_utilities.core.imports.RequirementCache.__bool__",
+            _shortcut_patch(RequirementCache.__bool__, ("neptune-client",), ("requirement",)),
+            "pytorch_lightning.loggers.neptune",
+        ),
+        (
+            "lightning_utilities.core.imports.RequirementCache.__bool__",
+            _shortcut_patch(RequirementCache.__bool__, ("jsonargparse[signatures]>=4.12.0",), ("requirement",)),
+            "pytorch_lightning.cli",
+        ),
+        (
+            "lightning_utilities.core.imports.module_available",
+            _shortcut_patch(module_available, ("fairscale.nn",)),
+            "pytorch_lightning.strategies",
+        ),
+        (
+            "lightning_utilities.core.imports.compare_version",
+            _shortcut_patch(compare_version, ("torch", operator.ge, "1.12.0")),
+            "pytorch_lightning.strategies.fully_sharded_native",
+        ),
+    ],
+    ids=["ProcessGroup", "neptune", "cli", "fairscale", "fully_sharded_native"],
+)
+def test_import_with_unavailable_dependencies(patch_name, new_fn, to_import, clean_import):
+    with mock.patch(patch_name, new=new_fn):
```

Comment:
How is this different to `test_imports` above?

Reply:
Put in more comments, please mark as resolved if you think the docstring explained it properly